packages/react/spec/auto/hooks/useFormFields.spec.tsx

```diff
@@ -0,0 +1,104 @@
+import { renderHook } from "@testing-library/react";
+import { useFormFields } from "../../../src/auto/AutoForm.js";
+import { useAutoFormMetadata } from "../../../src/auto/AutoFormContext.js";
+import type { ActionMetadata } from "../../../src/metadata.js";
+import { MockForm } from "../MockForm.js";
+
+describe("useFormFields hook", () => {
+  const getUseFormFieldsResult = (options: { include?: string[]; exclude?: string[] }) => {
+    const { result } = renderHook(
+      () => {
+        const { metadata: autoFormMetadata } = useAutoFormMetadata();
+        return useFormFields(autoFormMetadata, options);
+      },
+      {
+        wrapper: MockForm({ submit: jest.fn(), metadata }),
+      }
+    );
+
+    return result.current;
+  };
+
+  test("Returns the fields in the order of the action metadata when no options are used", () => {
+    const fields = getUseFormFieldsResult({});
+
+    expect(fields.map((field) => field.metadata.apiIdentifier)).toMatchInlineSnapshot(`
+        [
+          "stringField1",
+          "stringField2",
+          "stringField3",
+        ]
+      `);
+  });
+
+  test("Returns the fields in the order of the include options list", () => {
+    const include = ["stringField2", "stringField1", "stringField3"];
+
+    const fields = getUseFormFieldsResult({ include });
+    expect(fields.map((field) => field.metadata.apiIdentifier)).toEqual(include);
+  });
+
+  test("Excludes fields in the exclude list even if they are explicitly included", () => {
+    const fields = getUseFormFieldsResult({
+      include: ["stringField2", "stringField1", "stringField3"],
+      exclude: ["stringField2", "stringField3"],
+    });
```

Comment:
I think we should throw here -- the dev has given us conflicting information, I'd rather just be explicit and make them fix it?

Reply:
Updated to have an error if include/exclude have any overlap